File: magicheart/src/main/java/com/petrpopov/magicheart/data/Pulse.java
package com.petrpopov.magicheart.data;

import javax.validation.constraints.NotNull;

/**
 * User: petrpopov
 * Date: 30.09.13
 * Time: 11:53
 */

public class Pulse {

    @NotNull
    private Double value;

    @NotNull
    private Long timestamp;

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

    public Long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Long timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "Pulse{" +
                "value=" + value +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: magicclient/pulse.cpp
#include "pulse.h"

Pulse::Pulse(QObject *parent) :
    QObject(parent)
{
}

Pulse::Pulse(float value, qint64 timestamp, QObject *parent) : QObject(parent)
{
    this->value = value;
    this->timestamp = timestamp;
    this->time.setMSecsSinceEpoch(timestamp);
}

void Pulse::setValue(float value)
{
    this->value = value;
}

float Pulse::getValue()
{
    return this->value;
}

void Pulse::setTimestamp(qint64 timestamp)
{
    this->timestamp = timestamp;
    this->time.setMSecsSinceEpoch(timestamp);
}

qint64 Pulse::getTimestamp()
{
    return this->timestamp;
}

void Pulse::setTime(QDateTime time)
{
    this->time = time;
}

QDateTime Pulse::getTime()
{
    return this->time;
}
